R/basic.R

```diff
@@ -22,6 +25,15 @@ amqp_publish <- function(conn, body, exchange = "", routing_key = "",
   if (!inherits(conn, "amqp_connection")) {
     stop("`conn` is not an amqp_connection object")
   }
+  if (inherits(body, "character")) {
+    if (length(body) > 1) {
```

Comment:
This second check is not necessary; `rawToChar()` will issue a helpful warning message and fall back on reasonable behaviour here.